libraries/state_history/CMakeLists.txt

```diff
@@ -1,14 +1,28 @@
+add_executable(ship_abi_gen ship_abi_gen.cpp)
+target_include_directories( ship_abi_gen  PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
+target_link_libraries(ship_abi_gen abieos ${CMAKE_THREAD_LIBS_INIT})
+add_custom_command(OUTPUT ship_abi.cpp
+                   COMMAND ship_abi_gen > ship_abi.cpp
+                   DEPENDS ship_abi_gen)
+
+add_custom_target(ship_abi_app ALL
+   COMMAND echo "generate ship_abi.cpp"
+   DEPENDS ship_abi.cpp
+   VERBATIM
+)
+
```

Comment:
This breaks ("easy") cross compilation, which will be our strategy for ARM & non-linux builds going forward.

Reply:
Fixed with including a generated ABI in the source tree.